WordPress/Sniffs/Commenting/HooksInlineDocsSniff.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\Commenting;
+
+use WordPressCS\WordPress\AbstractFunctionRestrictionsSniff;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Class HooksMustHaveDocblockSniff
+ *
+ * @package WPCS\WordPressCodingStandards
+ */
+class HooksInlineDocsSniff extends AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Array of WordPress hook execution functions.
+	 *
+	 * @var array WordPress hook execution function name => filter or action.
+	 */
+	protected $hook_functions = array(
```

Comment:
We have a lot of arrays of functions names in the `Sniff` class that all of the Abstract (and therefore concrete) classes extend from - might be worth renaming the property to a bit more explicitly (`$hook_executing_functions`?) and then moving them to there so then other sniffs that need this collection can use them.

Reply:
Nice pointer. In 11e3a8fc, I removed my array. There is already a `$hookInvokeFunctions` array in the Sniff class.

The difference is the value of each key in the existing one is `true` whereas mine had `filter|action`. Originally, I was going to ensure that the "previously documented" check (`This filter is documented`) used `filter` or `action` correctly, but opted against being that nit-picky.

Since I wasn't used the key, let's just use what's already there.

Reply:
If the existing values were all `true`, then chances are the values weren't being used elsewhere. If there is benefit in having them be either 'filter' or 'action' so that _this_ sniff can check a little more finely (would you propose to throw a different violation if they didn't match?), then I'm all for that.

cc @jrfnl for consideration. Unresolving this comment so it doesn't get missed.

Reply:
I'm pretty sure that the values in the array aren't currently used, but a search of the codebase can verify this for sure.

I'm all for it to have `filter|action` as the values. The error messages could then potentially also use that term if it makes the error message clearer.